core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -238,6 +240,19 @@ public class GenericContainer<SELF extends GenericContainer<SELF>>
 
     private boolean hostAccessible = false;
 
+    private final Set<CreateContainerCmdModifier> createContainerCmdCustomizers = loadCreateContainerCmdCustomizers();
+
+    private Set<CreateContainerCmdModifier> loadCreateContainerCmdCustomizers() {
+        ServiceLoader<CreateContainerCmdModifier> containerCmdCustomizers = ServiceLoader.load(
+            CreateContainerCmdModifier.class
+        );
+        Set<CreateContainerCmdModifier> loadedCustomizers = new HashSet<>();
```

Comment:
FYI `HashSet` does not preserve the order of insertions. Consider using `LinkedHashSet` instead 👍